### ISSUE TITLE
Syslog support in python3 is broken

### ISSUE THREAD

**Issue by Joppie** [Supervisor/supervisor#1282]
## Problem

When running `supervisord` under python3 with a configuration that includes the program option `stdout_syslog = true`, messages do not show up in syslog.

## Versions tested

- 4.0.4
- ff7f18169bcc8091055f61279d0a63997d594148 (latest master at the time)

## Debugging

When running `supervisord` with `--nodaemon`, exception tracebacks reveal the problem:

```
Traceback (most recent call last):
  File "/home/staley/.local/lib/python3.7/site-packages/supervisor/loggers.py", line 372, in emit
    for line in message.rstrip('\n').split('\n'):
TypeError: a bytes-like object is required, not 'str'
```

`SyslogHandler.emit` is being passed a record that contains a message objects of type `bytes` not `str`. Passing `str` objects into the `rstrip` and `split` methods of a `byte` object is not allowed.

**Comment by mnaberez** [comment]
**Reproduce Instructions**

Create a config file `supervisord.conf`:

```
[supervisord]

[program:echo]
command = sh -c "echo hello"
stdout_syslog = true 
startsecs = 0
autostart = true
autorestart = false
```

Start `supervisord` in the foreground with that config:

```
$ supervisord -n -c supervisord.conf
2019-10-19 09:32:43,662 INFO supervisord started with pid 6076
2019-10-19 09:32:44,666 INFO spawned: 'echo' with pid 6079
Traceback (most recent call last):
  File ".../supervisor/loggers.py", line 372, in emit
    for line in message.rstrip('\n').split('\n'):
TypeError: a bytes-like object is required, not 'str'
2019-10-19 09:32:44,675 INFO success: echo entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
2019-10-19 09:32:45,677 INFO exited: echo (exit status 0; expected)
```

The exception above from [line 372 in `loggers.py`](https://github.com/Supervisor/supervisor/blob/ff7f18169bcc8091055f61279d0a63997d594148/supervisor/loggers.py#L372) is in `SyslogHandler`.  Tested on Python 2.7 and 3.6.  The exception only occurred on Python 3.6.

**Comment by mnaberez** [comment]
Fixed in #1284.
